src/core/connection.c

```diff
@@ -1315,6 +1319,10 @@ QuicConnOnShutdownComplete(
 
     if (Connection->State.ExternalOwner == FALSE) {
 
+        if (Connection->State.Started && !Connection->State.Connected) {
+            QuicPerfCounterIncrement(QUIC_PERF_COUNTER_CONN_HANDSHAKE_FAIL);
```

Comment:
In order to have a single place to increment this, let's just do it on free, if started but not connected.